### PR TITLE
scaladex: 1024- Add a fullScalaVersion field in Artifact (#1)

### PR DESCRIPTION
**Pull request by kannupriyakalra**
Issue- https://github.com/scalacenter/scaladex/issues/1024

- added fullScalaVersion in Artifact class
- added extractScalaVersion method
- added "full_scala_version" field in ArtifactTable
- added data migration file V20__add_full_scala_version.sql
- displayed the full Scala version to the user in the artifacts.scala.html template.
- updated tests

### CODE REVIEW

File: modules/server/src/main/scala/scaladex/server/service/ArtifactConverter.scala [hunk 1]
```diff
@@ -63,6 +56,22 @@ class ArtifactConverter(paths: DataPaths) extends LazyLogging {
...
+      dep.groupId == "org.scala-lang" &&
+      (dep.artifactId == "scala-library" || dep.artifactId == "scala3-library_3")
+    }
+
+    val sortedScalaDependencies = scalaDependencies.sortBy(_.artifactId)
+    val fullScalaVersion = scalaDependencies match {
+      case _ :: _ :: _ => Some(sortedScalaDependencies.last.version)
+      case head :: Nil => Some(head.version)
+      case _           => None
+    }
```
Comment (adpi2) [inline modules/server/src/main/scala/scaladex/server/service/ArtifactConverter.scala:70]: I think this is the same as:
```suggestion
    val fullScalaVersion = scalaDependencies.sortBy(_.artifactId).lastOption.map(_.version)
```

File: modules/template/src/main/twirl/scaladex/view/project/artifact.scala.html [hunk 2]
```diff
@@ -62,6 +62,7 @@ <h3>
               @if(artifact.resolver.isEmpty) {
                 @info("Files") { <a href="@artifact.mavenReference.repoUrl">View all</a> }
               }
+              @artifact.fullScalaVersion.map{ version => @info("Scala version") { @version.toString }}
```
Comment (adpi2) [inline modules/template/src/main/twirl/scaladex/view/project/artifact.scala.html:65]: I think we need to be more specific:

```suggestion
              @artifact.fullScalaVersion.map{ version => @info("Full Scala version") { @version.toString }}
```

**Review by adpi2 — APPROVED** [review]
Some suggestions. Apart from that it is looking quite good.

**Comment by adpi2** [comment]
Thanks @kannupriyakalra. I deployed this to https://index.scala-lang.org/. Let's wait a bit for new artifacts to come and we should see the `fullScalaVersion`.

**Comment by mzuehlke** [comment]
First artifact appeared: 

https://index.scala-lang.org/nationalarchives/da-metadata-schema/artifacts/da-metadata-schema/0.0.19?binary-version=_3

**Pull request merged**